### PR TITLE
INBOX-1140

### PR DESCRIPTION
**Pull request by pcatalini**
Add a layer to convert the test_id as an int

### CODE REVIEW

File: ns1/resource_datafeed.go [hunk 1]
```diff
@@ -88,11 +95,44 @@ func DataFeedDelete(d *schema.ResourceData, meta interface{}) error {
...
+
+//configAdapterIn adapts the configuration types
+func configAdapterIn(d *schema.ResourceData) error {
+	config := d.Get("config").(map[string]interface{})
+	if config != nil {
+		test_id := config["test_id"]
+		if test_id != nil {
+			test_id_int, err := strconv.Atoi(test_id.(string))
+			if err != nil {
+				return err
```
Comment (mburtless) [inline ns1/resource_datafeed.go:118]: I think we should wrap this error with a little more context about what failed, the default error is not going to be very helpful when troubleshooting a bad config in Terraform.

File: ns1/resource_datafeed.go [hunk 2]
```diff
@@ -35,24 +37,29 @@ func dataFeedResource() *schema.Resource {
...
 
-func resourceDataToDataFeed(d *schema.ResourceData) *data.Feed {
+func resourceDataToDataFeed(d *schema.ResourceData) (feed *data.Feed, e error) {
+	err := configAdapterIn(d)
 	return &data.Feed{
 		Name:     d.Get("name").(string),
 		SourceID: d.Get("source_id").(string),
 		Config:   d.Get("config").(map[string]interface{}),
-	}
+	}, err
```
Comment (mburtless) [inline ns1/resource_datafeed.go:52]: Typically we'd return a `nil` `*data.Feed` if an err is returned, no?  If the caller of this func doesn't check err for whatever reason, they could incorrectly try to use the returned pointer.

Comment (mburtless) [reply ns1/resource_datafeed.go:52]: Additionally, we probably want to wrap the error with a little more context about what failed.  The default error message is not going to be very useful for terraform users if they provided an invalid value for `test_id` and are troubleshooting what failed.

**Review by mburtless — CHANGES_REQUESTED** [review]
It would be good to get an acceptance test in `resource_datafeed_test` for this use case. I noted a couple of other things as well.

However, it seems to me that the validation and conversion of any datafeed config values would be a better fit for the Create and Update methods of DataFeedsService in the SDK.  Otherwise each consumer of the SDK will need to recreate this functionality as you are doing here.

File: ns1/resource_datafeed.go [hunk 3]
```diff
@@ -35,25 +39,39 @@ func dataFeedResource() *schema.Resource {
...
 		SourceID: d.Get("source_id").(string),
 		Config:   d.Get("config").(map[string]interface{}),
-	}
+	}, nil
 }
 
 // DataFeedCreate creates an ns1 datafeed
 func DataFeedCreate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*ns1.Client)
+<<<<<<< HEAD
```
Comment (mburtless) [inline ns1/resource_datafeed.go:63]: Looks like there are some artifacts left from a git merge in here that should be removed.

**Review by mburtless — CHANGES_REQUESTED** [review]

File: ns1/resource_datafeed.go (hunk 3 above)
Comment (pcatalini) [reply ns1/resource_datafeed.go]: Ops, my bad, I Forgot to save the merge before send

**Review by mburtless — APPROVED** [review]
Looks great!

**Pull request merged**